packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -321,18 +290,64 @@ class ProxyTarget {
         // Lookup the schema:
         const fieldKind = this.lookupFieldKind(key);
         // Make the childTargets:
-        const childTargets = mapCursorField(this.cursor, brand(key), (c) => new ProxyTarget(this.context, c));
-        return proxifyField(fieldKind, childTargets);
+        const childTargets = mapCursorField(this.cursor, brand(key), (c) => this.context.createTarget(c));
+        return proxifyField(this.context, fieldKind, childTargets);
     }
 
     /**
-     * @returns the type name of a node.
-     * It shall never be possible to call this for sequence fields as they are unwrapped into arrays.
+     * Sets value of a non-sequence field.
+     * This is correct only if sequence fields are unwrapped into arrays.
      */
-    public getTypeName(key: string): TreeSchemaIdentifier {
-        const childTypes = mapCursorField(this.cursor, brand(key), (c) => c.type);
-        assert(childTypes.length <= 1, 0x3c6 /* invalid non sequence */);
-        return childTypes[0];
+    public setValue(key: string, _value: unknown, typeName: TreeSchemaIdentifier): boolean {
```

Comment:
```suggestion
    public setValue(key: string, value: Value, typeName: TreeSchemaIdentifier): boolean {
```
We tend to reserve the use of leading underscores for variables that are not used in the implementation of the function, which is not the case here.

The type of `_value` should be `Value` from `src\tree\types.ts`